UnityProject/Assets/Scripts/Editor/Tools/AddressableToGameObjectTree.cs

```diff
@@ -0,0 +1,128 @@
+	/// </summary>
+	public class AddressableToGameObjectTree : EditorWindow
+	{
+		string path;
+		GameObject rootObject;
+		private string[] assetsPaths;
+
+		[MenuItem("Tools/Addressables to GameObject Tree Converter")]
+		private static void Init()
+		{
+			// Get existing open window or if none, make a new one:
+			AddressableToGameObjectTree window = (AddressableToGameObjectTree)GetWindow(typeof(AddressableToGameObjectTree));
+			window.titleContent.text = "Addressables to GameObject Tree Converter";
+			window.Show();
+		}
+
+		private void SelectPath()
+		{
+			string baseSTR = EditorUtility.OpenFolderPanel("Select folder to save prefabs", Application.dataPath, "prefabs");
+			if (baseSTR != "")
+			{
+				if (-1 == baseSTR.IndexOf(Application.dataPath, System.StringComparison.InvariantCultureIgnoreCase))
+				{
+					EditorUtility.DisplayDialog("Assets folder error", "Target folder must be any child directory to : \"" + Application.dataPath + "\"", "OK");
+					path = "";
+				}
+				else
+				{
+					path = baseSTR.Replace(Application.dataPath, "Assets");
+				}
+			}
+		}
+
+		private void ConvertToGameObjectRecursive(GameObject parentObject, string path)
+		{
+			string objectName = path.Split('/').Last();
+
+			GameObject gameObject = new GameObject(objectName);
+			gameObject.transform.parent = parentObject.transform;
+			Debug.Log($"Adding gameobject {objectName}");
+
+			// List of all prefabs exactly at path (without their file name)
+			List<string> prefabsAtPath = assetsPaths.Where(p => string.Join("/", p.Split('/').Reverse().Skip(1).Reverse()).Equals(path) && p.Contains(".prefab")).ToList();
+
+			//if (prefabsAtPath.Count > 0)
+			//{
+			//	AssetReferenceLibrary library = (AssetReferenceLibrary)gameObject.AddComponent(typeof(AssetReferenceLibrary));
+			//	library.AssetReferences = new List<AssetReference>();
+
+			//	// Seriously, that's the only way I found to list all prefabs in a given path.		
+			//	foreach (string prefabPath in prefabsAtPath)
+			//	{
+			//		library.AssetReferences.Add(new AssetReference(AssetDatabase.AssetPathToGUID(prefabPath)));
+			//		Debug.Log($"Adding reference to {prefabPath}");
+			//	}
+			//}
+
+			string[] subFolders = AssetDatabase.GetSubFolders(path);
+
+			foreach (string subFolder in subFolders)
+				ConvertToGameObjectRecursive(gameObject, subFolder);
+		}
+
+		void OnGUI()
+		{
+			EditorGUILayout.HelpBox("Choose the root path where the addressables are found", MessageType.Info);
+			EditorGUILayout.BeginHorizontal();
+			EditorGUILayout.LabelField("Addressables root folder:", GUILayout.MaxWidth(75.0f));
+
+			path = EditorGUILayout.TextField(path, EditorStyles.textField);
+
+			if (GUILayout.Button("...", GUILayout.MaxWidth(30.0f)))
+			{
+				SelectPath();
+			}
+
+			EditorGUILayout.EndHorizontal();
+
+			EditorGUILayout.Separator();
+
+			EditorGUILayout.HelpBox("Choose the GameObject that contains the tree of Asset References", MessageType.Info); ;
```

Comment:
Codacy found an issue: [Remove this empty statement.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)